repos/keg-components/src/components/list/sectionList.native.js

```diff
@@ -0,0 +1,394 @@
+        isScrollingRef.current,
+      ]
+    )
+  )
+}
+
+/**
+ * Helper hook to allow switching / scrolling to a section when a section change happens
+ * @param {boolean} doScrolling - Is automatic scrolling turned on
+ * @param {function} onSectionChange - Consumer Callback method for when a section changes
+ * @param {number} scrollOffset - Offset the final scroll position by some amount (px)
+ * @param {Object} sectionRefs - React ref of all Section divider components.
+ * @param {Object} listRef - React ref of the SectionList component
+ * @param {Object} isScrollingRef - React ref to track if we are scrolling
+ * @param {number} scrollCooldown - Amount of time to wait before allowing scrolling (ms)
+ *
+ * @returns {function} - Method to call when the current section is changed
+ */
+const useSectionChange = (
+  doScrolling,
+  onSectionChange,
+  scrollOffset,
+  setActiveSection,
+  sectionRefs,
+  listRef,
+  isScrollingRef,
+  scrollCooldown
+) => {
+  return useCallback(
+    index => {
+      // If there's no scrolling call the onSectionChange if it exists
+      if (!doScrolling) return checkCall(onSectionChange, index)
+
+      // Layout of the SectionHeader should be set in the useEffect hook
+      // of the SectionHeader component so it can be accessed here
+      const sectionData = sectionRefs.current[index]
+      const layout = sectionData?.layout
+
+      if (!layout)
+        return console.warn(`Section layout not correctly set`, sectionData)
+
+      // Update the is scrolling ref
+      // So other scrolling methods don't try to scroll
+      isScrollingRef.current = true
+
+      // Call the scroll method, to scroll to the section header based on it's layout
+      scrollList({
+        listRef,
+        animated: true,
+        behavior: 'smooth',
+        top: layout.top + scrollOffset,
+      })
+
+      // Call the passed in onSectionChange if it exists
+      checkCall(onSectionChange, index)
+
+      // Update the internal active section
+      setActiveSection(index)
+
+      // Wrap in a timeout to give it some cool down
+      // No way to track when scrolling stops, so this is the best we have
+      setTimeout(() => {
+        // Turn is scrolling back off
+        isScrollingRef.current = false
+      }, scrollCooldown)
+    },
+    [
+      scrollOffset,
+      scrollCooldown,
+      listRef.current,
+      onSectionChange,
+      setActiveSection,
+      sectionRefs.current,
+      isScrollingRef.current,
+    ]
+  )
+}
+
+/**
+ * Displays a section header for each section, adding a view to helper with tracking
+ * @param {Object} props
+ * @param {Object} props.index - Position of the section being rendered
+ * @param {Object} props.renderSectionHeader - Callback passed from the consumer
+ * @param {Object} props.sectionRefs - Ref Object to hold refs to the section header Dom Node
+ * @param {Array} props.sections - Groups of items to be displayed in the SectionList component
```

Comment:
looks like this should now be an object `section`?

Reply:
No, it's an array. What makes you say it's an object?

Reply:
I thought this was a typo since there's no `sections` in this function. just `section`
and from the usage on line `192` and `254` it looked like it's a singular item. unless im not understanding it right

Reply:
Ya, you are correct. I've updated it